src/kokkos/test/kokkos/HistoContainer_t.cc

```diff
@@ -0,0 +1,153 @@
+#include "KokkosCore/kokkosConfigCommon.h"
+#include "KokkosCore/kokkosConfig.h"
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <limits>
+#include <random>
+
+#include "KokkosCore/HistoContainer.h"
+
+template <typename T>
+void go() {
+  std::mt19937 eng;
+  std::uniform_int_distribution<T> rgen(std::numeric_limits<T>::min(), std::numeric_limits<T>::max());
+
+  constexpr int N = 12000;
+  Kokkos::View<T*,KokkosExecSpace> v_d("v_d",N);
+  typename Kokkos::View<T*,KokkosExecSpace>::HostMirror v_h("v_h",N);
+
+  constexpr uint32_t nParts = 10;
+  constexpr uint32_t partSize = N / nParts;
+//  uint32_t offsets[nParts + 1];
+
+  using Hist = HistoContainer<T, 128, N, 8 * sizeof(T), uint32_t, nParts>;
+  std::cout << "HistoContainer " << (int)(offsetof(Hist, off)) << ' ' << Hist::nbins() << ' ' << Hist::totbins() << ' '
+            << Hist::capacity() << ' ' << Hist::wsSize() << ' '
+            << (std::numeric_limits<T>::max() - std::numeric_limits<T>::min()) / Hist::nbins() << std::endl;
+
+  Kokkos::View<Hist*,KokkosExecSpace> h_d("h_d",1);
+  typename Kokkos::View<Hist*,KokkosExecSpace>::HostMirror h_h("h_h",1);
+
+  Kokkos::View<uint32_t*,KokkosExecSpace> off_d("off_d",nParts + 1);
+  Kokkos::View<uint32_t*,KokkosExecSpace>::HostMirror off_h("off_h",nParts + 1);
+
+  for (int it = 0; it < 5; ++it) {
+    printf("it = %10d\n",it);
+    off_h(0) = 0;
+    for (uint32_t j = 1; j < nParts + 1; ++j) {
+      off_h(j) = off_h(j-1) + partSize - 3 * j;
+      assert(off_h(j) <= N);
+    }
+
+    if (it == 1) {  // special cases...
+      off_h(0) = 0;
+      off_h(1) = 0;
+      off_h(2) = 19;
+      off_h(3) = 32 + off_h(2);
+      off_h(4) = 123 + off_h(3);
+      off_h(5) = 256 + off_h(4);
+      off_h(6) = 311 + off_h(5);
+      off_h(7) = 2111 + off_h(6);
+      off_h(8) = 256 * 11 + off_h(7);
+      off_h(9) = 44 + off_h(8);
+      off_h(10) = 3297 + off_h(9);
+    }
+    Kokkos::deep_copy(off_d,off_h);
```

Comment:
Please pass the execution space object, e.g.
```suggestion
    Kokkos::deep_copy(KokkosExecSpace(), off_d,off_h);
```
so that the serial build of the unit test can be run on a machine without GPU.

Reply:
Added this in all the unit tests